apps/test/unit/code-studio/components/progress/LessonLockDialog/LessonLockDialogTest.jsx

```diff
@@ -0,0 +1,203 @@
+    wrapper.update();
+
+    getStudentRows(wrapper).forEach(row => {
+      expect(row.props().lockStatus).to.equal(LockStatus.Locked);
+    });
+
+    lessonLockDataApi.useGetLockState.restore();
+  });
+
+  it('clicking "Show answers" sets all statuses to ReadOnlyAnswers', () => {
+    sinon.stub(lessonLockDataApi, 'useGetLockState').returns({
+      loading: false,
+      serverLockState: [
+        {name: 'fakeName1', lockStatus: LockStatus.Editable},
+        {name: 'fakeName2', lockStatus: LockStatus.Editable}
+      ]
+    });
+
+    const wrapper = mount(<LessonLockDialog {...MINIMUM_PROPS} />);
+
+    getStudentRows(wrapper).forEach(row => {
+      expect(row.props().lockStatus).to.equal(LockStatus.Editable);
+    });
+
+    const showAnswersButton = wrapper.find('button').at(2);
+    expect(showAnswersButton.text() === 'Show answers');
+    showAnswersButton.simulate('click');
+    wrapper.update();
+
+    getStudentRows(wrapper).forEach(row => {
+      expect(row.props().lockStatus).to.equal(LockStatus.ReadonlyAnswers);
+    });
+
+    lessonLockDataApi.useGetLockState.restore();
+  });
+
+  describe('viewSection callback', () => {
+    beforeEach(() => sinon.stub(window, 'open'));
+    afterEach(() => window.open.restore());
+
+    it('opens a window to the section assessments page', () => {
+      const wrapper = mount(<LessonLockDialog {...MINIMUM_PROPS} />);
+
+      const viewSectionButton = wrapper.find('button').at(4);
+      expect(viewSectionButton.text() === 'View section');
+      viewSectionButton.simulate('click');
+      wrapper.update();
+
+      expect(window.open).to.have.been.calledOnce.and.calledWith(
+        `/teacher_dashboard/sections/${fakeSectionId}/assessments`
+      );
+    });
+  });
+
+  it('handleSave calls saveLockState, refetchSectionLockStatus and handleClose', async () => {
+    const initialLockStatus = [
+      {name: 'fakeName1', lockStatus: LockStatus.Editable},
+      {name: 'fakeName2', lockStatus: LockStatus.Editable}
+    ];
+    sinon.stub(lessonLockDataApi, 'useGetLockState').returns({
+      loading: false,
+      serverLockState: initialLockStatus
+    });
+    const lessonLockSaveStub = sinon
+      .stub(lessonLockDataApi, 'saveLockState')
+      .returns(new Promise(resolve => resolve()));
+    const refetchStub = sinon.stub().returns(new Promise(resolve => resolve()));
+    const handleCloseSpy = sinon.spy();
+
+    const wrapper = mount(
+      <LessonLockDialog
+        {...MINIMUM_PROPS}
+        refetchSectionLockStatus={refetchStub}
+        handleClose={handleCloseSpy}
+      />
+    );
+
+    const lockLessonButton = wrapper.find('button').at(1);
+    expect(lockLessonButton.text() === 'Lock lesson');
+    lockLessonButton.simulate('click');
+    wrapper.update();
+
+    const saveButton = wrapper.find('button').at(6);
+    expect(saveButton.text() === 'Save');
+    saveButton.simulate('click');
+    wrapper.update();
+
+    await setTimeout(() => {}, 50);
```

Comment:
Is there any concern that these timeouts can sauce flakiness?

Reply:
I'm hoping the timeouts will help avoid flakiness, however that's definitely worth monitoring